connectors/bitbucket-server/src/main/java/com/vmware/connectors/bitbucket/server/BitbucketServerController.java

```diff
@@ -0,0 +1,484 @@
+                pullRequest,
+                baseUrl);
+
+        return performBitbucketServerAction(baseUrl, authHeader, pullRequest, BitbucketServerAction.MERGE);
+    }
+
+    @PostMapping(
+            path = "/api/v1/{projectKey}/{repositorySlug}/{pullRequestId}/decline",
+            consumes = MediaType.APPLICATION_FORM_URLENCODED_VALUE
+    )
+    public Single<ResponseEntity<String>> decline(final BitbucketServerPullRequest pullRequest,
+                                                @RequestHeader(AUTH_HEADER) final String authHeader,
+                                                @RequestHeader(BASE_URL_HEADER) final String baseUrl) {
+
+        logger.debug("Decline ACTION for bitbucket server pull request: {}, baseURL: {}",
+                pullRequest,
+                baseUrl);
+
+        return performBitbucketServerAction(baseUrl, authHeader, pullRequest, BitbucketServerAction.DECLINE);
+    }
+
+    @PostMapping(
+            path = "/api/v1/{projectKey}/{repositorySlug}/{pullRequestId}/comments",
+            consumes = MediaType.APPLICATION_FORM_URLENCODED_VALUE
+    )
+    public Single<ResponseEntity<String>> comments(final BitbucketServerPullRequest pullRequest,
+                                                  @RequestParam(COMMENT_PARAM_KEY) final String comment,
+                                                  @RequestHeader(AUTH_HEADER) final String authHeader,
+                                                  @RequestHeader(BASE_URL_HEADER) final String baseUrl) {
+
+        logger.info("Comment ACTION for bitbucket server pull request: {}, baseURL: {}",
+                pullRequest,
+                baseUrl);
+
+        final HttpHeaders headers = buildHeaders(authHeader);
+        headers.add(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE);
+
+        final URI uri = UriComponentsBuilder.fromHttpUrl(baseUrl + "/rest/api/1.0/projects/{PROJECT_KEY}/repos/{REPOSITORY_SLUG}/pull-requests/{PULL_REQUEST_ID}/comments")
+                .buildAndExpand(
+                        ImmutableMap.of(
+                                PROJECT_KEY, pullRequest.getProjectKey(),
+                                REPOSITORY_SLUG, pullRequest.getRepositorySlug(),
+                                PULL_REQUEST_ID, pullRequest.getPullRequestId()
+                        )
+                ).toUri();
+
+        final BitbucketServerComment bitBucketServerComment = new BitbucketServerComment(comment);
+        final ListenableFuture<ResponseEntity<String>> response = this.rest.exchange(
+                uri,
+                HttpMethod.POST,
+                new HttpEntity<>(bitBucketServerComment, headers),
+                String.class);
+        return Async.toSingle(response);
+    }
+
+    @GetMapping("/test-auth")
+    @SuppressWarnings("PMD.JUnit4TestShouldUseTestAnnotation") // PMD assumes the method as Junit Test if the API url has 'test'.
```

Comment:
You can get rid of this if you rename the method (don't start w/ test, so authTest or verifyAuth or something).

Reply:
Renamed the method name to ```verifyAuth```